### PR TITLE
Semver: specify VersionNumber's follow v2-rc2 specifically

### LINKED ISSUE #53502: comparison bug in `Version`
Since semVer 2.0.0, they have dropped the `build` part for comparison completely, so both of these are wrong:

```julia
julia> v"1.2.3-pre.3+build.3" < v"1.2.3-pre.3+build.4"
true

julia> v"1.2.3-pre.3+build.4" == v"1.2.3-pre.3+build.3"
false
```

ref: https://semver.org/spec/v2.0.0.html#spec-item-10

spotted by @avx-ctan

### PR DESCRIPTION
**Pull request by ericphanson**
Since v2 has changed from v2-rc2 by saying build numbers MUST not be taken into account in precedence.

I don't think changing VersionNumber's ordering in a minor release of Julia is acceptable, but since the semver landing page is now v2 rather than v2-rc2, we should update the link.

closes https://github.com/JuliaLang/julia/issues/53502

### CODE REVIEW

**Comment by Moelf** [comment]
we explain what literal version numbers are here: https://github.com/JuliaLang/julia/blob/c06662a4f8d3fb277a30df82f2a012739f3c84d6/doc/src/manual/strings.md?plain=1#L1141

would be useful to also add a RC2 note there

File: doc/src/manual/strings.md [hunk 1]
```diff
@@ -1142,7 +1142,7 @@ some confusion regarding the matter.
 
 Version numbers can easily be expressed with non-standard string literals of the form [`v"..."`](@ref @v_str).
 Version number literals create [`VersionNumber`](@ref) objects which follow the
-specifications of [semantic versioning](https://semver.org/),
+specifications of [semantic versioning](https://semver.org/spec/v2.0.0-rc.2.html),
```
Comment (ericphanson) [inline doc/src/manual/strings.md:1145]: @Moelf that is already updated here

Comment (Moelf) [inline doc/src/manual/strings.md:1145]:
```suggestion
specifications of [semantic versioning 2.0.0-RC2](https://semver.org/spec/v2.0.0-rc.2.html),
```

I mean if someone reading it they don't see RC2

Comment (ericphanson) [reply doc/src/manual/strings.md:1145]: it seems they use lowercase:
```suggestion
specifications of [semantic versioning 2.0.0-rc2](https://semver.org/spec/v2.0.0-rc.2.html),
```

**Pull request merged**